framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -1051,13 +1051,25 @@ public Void visitVariable(VariableTree node, Void p) {
      * @param modifiersTree the modifiers sub-tree of node
      */
     private void warnAboutTypeAnnotationsTooEarly(Tree node, ModifiersTree modifiersTree) {
-        if (node.getKind() == Tree.Kind.VARIABLE
-                && TreeUtils.elementFromDeclaration((VariableTree) node).getKind()
-                        == ElementKind.ENUM_CONSTANT) {
-            // Enums constants are "public static final" by default, so the annotation always
-            // appears to be before public.
-            return;
+
+        // Don't issue warnings about compiler-inserted modifiers.
+        // These two tests could be made more precise and catch some user errors in enum constants
+        // and in try-with-resources declarations, but it doesn't seem worth the effort.
+        if (node.getKind() == Tree.Kind.VARIABLE) {
+            if (TreeUtils.elementFromDeclaration((VariableTree) node).getKind()
+                    == ElementKind.ENUM_CONSTANT) {
+                // Enum constants are "public static final" by default, so the annotation always
+                // appears to be before "public".
+                return;
+            }
+            Tree parent = getCurrentPath().getParentPath().getLeaf();
```

Comment:
I'm always a bit worried about using Paths, because they can be costly.
I think you can simplify this significantly:
- determine the variable kind once `kind = TreeUtils.elementFromDeclaration((VariableTree) node).getKind()`
- if kind is ENUM_CONSTANT it's an enum constant
- if kind is RESOURCE_VARIABLE it's a try resource

So you don't need a path and the two checks are more symmetric.

Reply:
Yes, that is better.  I made that change.